images/scorecard-test-kuttl/cmd/test-kuttl/main.go

```diff
@@ -74,9 +74,22 @@ func main() {
 }
 
 func getTestStatus(tc []*Testcase) (s v1alpha3.TestStatus) {
+
+	logMessage := getKuttlLogs()
+
+	// report the kuttl logs when kuttl tests can not be run
+	// (e.g. RBAC is not sufficient)
+	if len(tc) == 0 {
+		r := v1alpha3.TestResult{}
+		r.Log = logMessage
+		s.Results = append(s.Results, r)
+		return s
+	}
+
 	for i := 0; i < len(tc); i++ {
 		r := v1alpha3.TestResult{}
 		r.Name = tc[i].Name
+		r.Log = logMessage
```

Comment:
This is putting the same log message in every single test result right? I'm not familiar with kuttl's verbosity when things run normally, but I'm wondering if we should only show the logs when `len(tc) == 0`?

Reply:
correct, it goes in every test result....kuttl has a single stderr and stdout for all its tests, we could only add that log output when len(tc) == 0 but then you lose the ability to view the logs altogether which can be useful.  I was wondering about adding a Log field to the scorecard API at the TestResult level to handle this sort of case but that requires a scorecard API change.

Reply:
> you lose the ability to view the logs altogether which can be useful

If we did it this way, wouldn't it mean seeing the logs when the `kuttl` process fails and not seeing the logs when kuttl returns test case results? If so, that seems pretty reasonable at least as a first step.

Is the stdout or stderr useful to see when tests run normally, or only when kuttl bombs out?
Also, is there any possibility of parsing the kuttl's logs and splitting for each test case?

Reply:
Alternatively, will logs be different per test? Can these logs be collected by `tc[i].Name` and added to the appropriate result?

Reply:
``
	Log:
		=== RUN   kuttl
		    harness.go:435: starting setup
		    harness.go:247: running tests using configured kubeconfig.
		    harness.go:335: running tests
		    harness.go:73: going to run test suite with timeout of 120 seconds for each step
		=== RUN   kuttl/harness
		=== RUN   kuttl/harness/list-pods
		=== PAUSE kuttl/harness/list-pods
		=== CONT  kuttl/harness/list-pods
		    logger.go:42: 20:37:15 | list-pods | Skipping creation of user-supplied namespace: default
		    logger.go:42: 20:37:15 | list-pods/0-pod | starting test step 0-pod
		    logger.go:42: 20:37:15 | list-pods/0-pod | Pod:default/pod-1 created
		    logger.go:42: 20:37:15 | list-pods/0-pod | test step completed 0-pod
		    logger.go:42: 20:37:15 | list-pods | skipping kubernetes event logging
		    logger.go:42: 20:37:15 | list-pods | Skipping deletion of user-supplied namespace: default
		=== CONT  kuttl
		    harness.go:376: run tests finished
		    harness.go:480: cleaning up
		    harness.go:532: removing temp folder: ""
		--- PASS: kuttl (0.87s)
		    --- PASS: kuttl/harness (0.00s)
		        --- PASS: kuttl/harness/list-pods (0.36s)
		PASS

thats an example of a 'passing' log, not much value perhaps, my vote would be to just show this log when there is a failure (e.g. no test cases).  If it turns out this log output is useful, we could add that on as a future feature later on.

Reply:
I updated the code to only populate the Log when a kuttl error scenario is detected (e.g. len(tc) == 0).